src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1346,7 +1375,7 @@ internal virtual HttpResponseMessage GetResponse(HttpClient client, HttpRequestM
                             IEnumerator<string> enumerator = retryAfter.GetEnumerator();
                             if (enumerator.MoveNext())
                             {
-                                retryIntervalInSeconds = Convert.ToInt32(enumerator.Current);
+                                retryIntervalInSeconds = (double)Convert.ToInt32(enumerator.Current);
```

Comment:
```suggestion
                                retryIntervalInSeconds = Convert.ToSingle(enumerator.Current);
```

Reply:
Since the value specified in the Retry-After header is a non-negative decimal integer, I retained the use of ToInt32 thinking that conversion to floating point should not be attempted.
Should the implementation on the Cmdlet side be more forgiving?
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After

Reply:
```suggestion
                                retryIntervalInSeconds = Convert.ToInt32(enumerator.Current);
```
You were right

Reply:
```suggestion
                                retryIntervalInSeconds = Convert.ToInt32(enumerator.Current);
```
You are right this should stay int